pkg/resource/deploy/step_generator.go

```diff
@@ -1258,6 +1273,10 @@ func (sg *stepGenerator) generateStepsFromDiff(
 			if goal.DeleteBeforeReplace != nil {
 				deleteBeforeReplace = *goal.DeleteBeforeReplace
 			}
+			// If Autonaming settings had no randomness in the name, we must delete before create a replacement.
+			if autonamingDeleteBeforeCreate {
```

Comment:
I feel like we want to call this `autonamingDeleteBeforeReplace` and similarly throughout

Reply:
I never understood the `deleteBeforeReplace` name and we already have `DeleteBeforeCreate` a few lines above. But I agree my naming is confusing and inconsistent with the variable that we re-assign in the if statement, so I'll rename.